File: 3959-check-good-integer/3959-check-good-integer.cpp
class Solution {
public:
    bool checkGoodInteger(int n) {
        string s = to_string(n);
        int d_s = 0 , s_s = 0;
        for(int el : s){
            d_s += el - '0';
            s_s += (el - '0') * (el - '0');
        }
        return (s_s - d_s)>= 50;
    }
};

File: 3959-check-good-integer/3959-check-good-integer.java
class Solution {
    public boolean checkGoodInteger(int n) {
        String s = String.valueOf(n);
        int s_s = 0 , d_s = 0;
        for(char ch : s.toCharArray()){
            s_s += (ch - '0') * (ch - '0');
            d_s += (ch - '0');
        }
        return (s_s - d_s)>=50;
    }
}